### ISSUE TITLE
Feature: Clear Ages

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#193]
Good suggestion From Rob B.
I thought we had this in there, but I guess not!

Maybe we need two? 
a) clear the current session?
b) clear all sessions?  Are you sure?

**Comment by mpogue2** [comment]
@danlyke This one is yours, should you choose to accept it....no rush, since it's a nice-to-have feature, and not a critical bug.

**Comment by danlyke** [comment]
So do we want to do this as something that actually deletes the play history, or that just puts a "cleared" flag in the play history? The latter would make sure that we wouldn't nuke anything out that'd cause a problem with #81, on the other hand this could also be a way to compact the database when it gets too huge (ie: a couple decades of play history...)

**Comment by mpogue2** [comment]
I don't think the database grows that fast, so just keeping a "fence", something like "only look at plays that happened after this date, and ignore earlier ones" might work.  Then, we just need a fence for each session.  Doesn't have to be in the database, unless you want to plan ahead for user-defined sessions.  Could just be a bunch of named preferences.

Or, if you want to get fancy, then a column in the DB that has a key "session name" and a value "date that this session was last cleared" would also work.  That way, we can still do statistics on when songs were played (how often, etc.), without actually having to delete playback events just for the Clear Ages feature.

I think it's your call, depending on how fancy you want to get here!

**Comment by mpogue2** [comment]
Feedback from Rob B: "Still unconvinced about the “Age: column showing days since last use. Have you considered just using a flag showing use and a reset for the column. This will enable the caller to cycle his music by just observing a flag rather than having to review the elapsed time which can be time consuming. I personally like to know if I have simply used a song not that was used 28 days ago."

@danlyke What do you think?  Could be a UI option to show check or not vs Age.  Still need the clear age option in any case (menu option?).  Should it clear the DB, or just set a fence?

**Comment by danlyke** [comment]
I think it should set a fence.

I'm okay with an option to show a checkbox or not...

I'd toss this in fast, but I've run into a problem with the web view in
the current branch: Apparently support for full Qt5.9 is less robust in
Ubuntu than I thought. I have a potential work-around, using a web view
called "Oxide", but I haven't gotten there yet.

**Comment by mpogue2** [comment]
OK, I've added a fence.  Note that there's only one fence right now, and it affects all Sessions.  If you click on Session > Clear Recent, the fence is set to NOW.  At some point, we might want to have one fence per Session (presumably the current Session, but then do we need Clear Recent for All Sessions?).  Do we also need "Clear Recents older than 1 week"?  We can open up a separate issue for these, if there's interest.

The Recent column and the fence went in as 665c502844669a5e04ae439bc6ccdc93683735b7 .
Closing for now.
